extensions/panache/panacheql/.gitignore

```diff
@@ -0,0 +1,3 @@
+gen/**
+**/internal/gen/**
+/src/main/antlr4/io/quarkus/panacheql/internal/HqlLexer.tokens
```

Comment:
Hmmm, is this needed? If so, we probably need to have it in the root .gitignore. We don't want to have rules all over the place.

Reply:
The files appeared in my workspace, so I assumed it would be a problem for everyone and I'd just pop it in with the other small change, rather than making a new PR just for the .gitignore.

I checked before I put the .gitignore in to decide what our pattern was, and decided it was to use project-level .gitignores:

```
holly@hcummins-mac quarkus % find . -name .gitignore
./independent-projects/arc/.gitignore
./integration-tests/elasticsearch-rest-client/.gitignore
./integration-tests/funqy-google-cloud-functions/.gitignore
./integration-tests/google-cloud-functions-http/.gitignore
./integration-tests/gradle/.gitignore
./extensions/panache/panacheql/.gitignore
./.gitignore
./devtools/cli/.idea/.gitignore
./devtools/gradle/.gitignore
./devtools/project-core-extension-codestarts/target/classes/gradle-wrapper/.gitignore
./coverage-report/.gitignore
```

The disadvantage of handling every ignore in the root is you end up with the root gitignore having detailed knowledge of child project paths, and if you move a project around, then all the ignored files pop back up. The advantage of having a single file is that it's a single file. 

I assumed the `gen` folders would be popping up for everyone, which is why I fixed it, but thinking about it, it may only show for people who happen to have the IntelliJ Antlr plugin installed. That's a smaller group - but the bad consequences of ignoring the file are also pretty low.